lambdas/data-migration1/tests/test-providers.js

```diff
@@ -0,0 +1,363 @@
+const cryptoRandomString = require('crypto-random-string');
+const omit = require('lodash/omit');
+const path = require('path');
+const test = require('ava');
+
+const Provider = require('@cumulus/api/models/providers');
+const Rule = require('@cumulus/api/models/rules');
+const KMS = require('@cumulus/aws-client/KMS');
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const {
+  createBucket,
+  putFile,
+  recursivelyDeleteS3Bucket,
+} = require('@cumulus/aws-client/S3');
+const { S3KeyPairProvider } = require('@cumulus/common/key-pair-provider');
+const { getKnexClient, localStackConnectionEnv } = require('@cumulus/db');
+
+const {
+  migrateProviderRecord,
+  migrateProviders,
+} = require('../dist/lambda/providers');
+const { RecordAlreadyMigrated } = require('../dist/lambda/errors');
+
+const testDbName = `data_migration_1_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+
+const generateFakeProvider = (params) => ({
+  id: cryptoRandomString({ length: 10 }),
+  globalConnectionLimit: 1,
+  protocol: 'http',
+  host: `${cryptoRandomString({ length: 10 })}host`,
+  port: 80,
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+  username: `${cryptoRandomString({ length: 5 })}user`,
+  password: `${cryptoRandomString({ length: 5 })}pass`,
+  encrypted: false,
+  privateKey: 'key',
+  cmKeyId: 'key-id',
+  certificateUri: 'uri',
+  ...params,
+});
+
+let providersModel;
+let rulesModel;
+
+test.before(async (t) => {
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir: `${path.join(__dirname, '..', '..', 'db-migration', 'dist', 'lambda', 'migrations')}`,
+    },
+  });
+
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  await putFile(
+    process.env.system_bucket,
+    `${process.env.stackName}/crypto/public.pub`,
+    require.resolve('@cumulus/test-data/keys/s3_key_pair_provider_public.pub')
+  );
+
+  await putFile(
+    process.env.system_bucket,
+    `${process.env.stackName}/crypto/private.pem`,
+    require.resolve('@cumulus/test-data/keys/s3_key_pair_provider_private.pem')
+  );
+
+  const createKeyResponse = await KMS.createKey();
+  process.env.provider_kms_key_id = createKeyResponse.KeyMetadata.KeyId;
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  rulesModel = new Rule();
+  await rulesModel.createTable();
+
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
```

Comment:
We shouldn't have to create the database and run migrations in the test, it should be handled by `npm run db:local:migrate`.

Reply:
This goes back to a pattern from @Jkovarik's previous PR. I believe that his argument was that each test should be using a separate database and thus needs the migrations applied to that database.

Reply:
I'm not sure why each test would need its own database, though, if what we are testing is a data migration and not a schema migration.